lib/Common/Memory/SectionAllocWrapper.cpp

```diff
@@ -877,61 +877,53 @@ LPVOID PreReservedSectionAllocWrapper::Alloc(LPVOID lpAddress, size_t dwSize, DW
 BOOL
 PreReservedSectionAllocWrapper::Free(LPVOID lpAddress, size_t dwSize, DWORD dwFreeType)
 {
-    {
-        AutoCriticalSection autocs(&this->cs);
+    AutoCriticalSection autocs(&this->cs);
 
-        if (dwSize == 0)
-        {
-            Assert(false);
-            return FALSE;
-        }
+    if (dwSize == 0)
+    {
+        Assert(false);
+        return FALSE;
+    }
 
-        if (this->preReservedStartAddress == nullptr)
-        {
-            Assert(false);
-            return FALSE;
-        }
+    if (this->preReservedStartAddress == nullptr)
+    {
+        Assert(false);
+        return FALSE;
+    }
 
-        Assert(dwSize % AutoSystemInfo::PageSize == 0);
-        BOOL success = TRUE;
-        if((dwFreeType & MEM_DECOMMIT) == MEM_DECOMMIT)
-        {
-            LPVOID localAddr = AllocLocal(lpAddress, dwSize);
-            if (localAddr == nullptr)
-            {
-                success = FALSE;
-            }
-            else
-            {
-                ZeroMemory(localAddr, dwSize);
-                FreeLocal(localAddr);
-                DWORD oldFlags = NULL;
-                success = VirtualProtectEx(this->process, lpAddress, dwSize, PAGE_NOACCESS, &oldFlags);
-            }
-        }
-        size_t requestedNumOfSegments = dwSize / AutoSystemInfo::Data.GetAllocationGranularityPageSize();
-        Assert(requestedNumOfSegments <= MAXUINT32);
+    Assert(dwSize % AutoSystemInfo::PageSize == 0);
+    LPVOID localAddr = AllocLocal(lpAddress, dwSize);
+    if (localAddr == nullptr)
+    {
+        return FALSE;
+    }
+    ZeroMemory(localAddr, dwSize);
```

Comment:
>ZeroMemory(localAddr, dwSize); [](start = 4, length = 30)

Could this be moved to worker thread and use non-temporal write to avoid unnecessary working set increasing?